code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm

```diff
@@ -410,7 +421,6 @@
 	target_xeno.hud_update_banished()
 	target_xeno.lock_evolve = TRUE
 	user_xeno.hive.banished_ckeys[target_xeno.name] = target_xeno.ckey
-	addtimer(CALLBACK(src, PROC_REF(remove_banish), user_xeno.hive, target_xeno.name), 30 MINUTES)
```

Comment:
Revert.

This is intended and it will only remove the ckey of the xeno in the hive list, not alter the banished state on the xeno should their mob still somehow survive 30 mins later. It will make them unable to be readmitted after that point - but reasonable IMO if they've managed to maintain this state for over 30 minutes and not die or be readmitted or for queen that banished to die.

The issue I assume is that you're attaching a timer to an ability (src) that you are removing. So attach the timer to the hive instead. We just need to ensure that

Reply:
There is also an edge case where if a person is banished, queen dies resulting in unbanishment, then they join the hive again, and should next queen then banish them again before that 30 minutes, they will get into that un-readmittable state again. But this is extremely situational that at most is punishing the player additionally for causing multiple queens to banish them multiple times in a 30 minute window...